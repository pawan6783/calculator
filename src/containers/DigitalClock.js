import React, {useState} from 'react';
import '../stylesheets/DigitalClock.css';
import Moment from 'react-moment';

const DigitalClock = props => {

    const [timeState,setTimeState] = useState({
        time: new Date().toLocaleTimeString(),
        date: new Date()
    });
    const [flagState,setFlagState] = useState({showMe: false});

    const updateTime = () => {
       
        console.log(flagState.showMe);
        setTimeState({
            time: new Date().toLocaleTimeString(),
            date: new Date().toLocaleDateString()
        });
    }

    const onChangeHandler = () => {
       console.log("inside onchangehandler");
        setFlagState({
            showMe: !flagState.showMe
        });
        
    }

    setInterval(updateTime,1000);

    return(
        <div>
            <h1>Digital Clock</h1>
            <label className="switch">
                <input type="checkbox" onChange = {onChangeHandler}/>
                <span className="slider round"></span>
            </label>
            <h1>{timeState.time}</h1>
            <h4>{
                 flagState.showMe?
                 <Moment format = "MMMM Do YYYY dddd">
                {timeState.date}
                </Moment> : null
                }
            </h4>
        </div>
    );
}

export default DigitalClock;