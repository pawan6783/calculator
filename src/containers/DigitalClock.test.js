import React from 'react';
import {configure, shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import DigitalClock from './DigitalClock';

configure({ adapter: new Adapter()});

describe("Testing of <DigitalClock/> component", () => {


    it("should match `DigitalClock` component with snapshot with no props", () => {
        //console.log(wrapper.debug());
        const wrapper = shallow(<DigitalClock></DigitalClock>);
        expect(wrapper).toMatchSnapshot();
    });

    it("should return `onChangeHandler` method on toggle", () => {

        const onChangeHandler = jest.fn();
        const wrapper = shallow(<DigitalClock onChange={onChangeHandler}></DigitalClock>)
        wrapper.find('input').simulate('onChange');
        expect(onChangeHandler).toBeDefined();
    });

    it("should return  `h1` element with text", () => {

        const wrapper = mount(<DigitalClock></DigitalClock>);
        expect(wrapper.find('h1').first().text()).toEqual('Digital Clock');
        wrapper.unmount();
    });
});