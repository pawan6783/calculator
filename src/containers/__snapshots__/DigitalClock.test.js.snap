// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing of <DigitalClock/> component should match \`DigitalClock\` co with text 1`] = `ShallowWrapper {}`;

exports[`Testing of <DigitalClock/> component should match \`DigitalClock\` component with snapshot  1`] = `ShallowWrapper {}`;

exports[`Testing of <DigitalClock/> component should match \`DigitalClock\` component with snapshot with no props 1`] = `ShallowWrapper {}`;

exports[`Testing of <DigitalClock/> component should match \`DigitalComponent\` element with text 1`] = `ShallowWrapper {}`;

exports[`Testing of <DigitalClock/> component should return \`\` element with text 1`] = `ShallowWrapper {}`;

exports[`Testing of <DigitalClock/> component should return one \`h1\` element with text 1`] = `ShallowWrapper {}`;
